libs/dal/src/repositories/message/message.repository.ts

```diff
@@ -685,12 +685,16 @@ export class MessageRepository extends BaseRepository<MessageDBModel, MessageEnt
       _environmentId: string;
     } & Partial<Omit<MessageEntity, 'transactionId'>>
   ) {
-    const res = await this.MongooseModel.find({
-      transactionId: {
-        $in: query.transactionId,
+    const res = await this.MongooseModel.find(
+      {
+        transactionId: {
+          $in: query.transactionId,
+        },
+        _environmentId: query._environmentId,
       },
-      _environmentId: query._environmentId,
-    })
+      {},
+      { sort: { createdAt: -1 } }
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** ‏ `messages` collection has 116_756_136 documents on US prod, we might consider adding index on `createdAt` (or possibly compound index with the other fields)